### DISCUSSION TITLE
카드 정보 form 데이터를 카드 등록 완료 페이지로 어떻게 넘겨줄 수 있을까?

### DISCUSSION THREAD

**Discussion by Tanney-102** [woowacourse/react-payments#45] [토론장]
1. 상위 컴포넌트에서 state로 관리한다.
2. 세션 스토리지나 히스토리에 저장
3. 카드정보를 submit할때 서버에 정보를 저장하고 id만 1,2방법 중 하나로 넘긴다

**Comment by hchayan** [comment]
만약 라우터로 넘길때는 history.push할때 안에 state를 담아 넘길수도 있어요!
react-router에 있는 useHistory와 useLocation에 대한 정보를 찾아보면 좋을거같아요
```
// 예제
history.push({
            pathname: '/cardComplete',  // 이동할 페이지 라우터
            search: '?query=userName',  // 에 넣어줄 쿼리
            state: { cardFormInfo: cardFormInfo },  // 넘겨줄 state
          });
...

**Reply by Tanney-102** [reply]
넵 근데! 카드 정보는 오히려 바로 서버로 보내 관리하는게 나아보이기도 하네요ㅜㅜ 뭐가 가장 적절할지 고민입니다 ㅎㅎ

**Reply by bigsaigon333** [reply]
> 만약 라우터로 넘길때는 history.push할때 안에 state를 담아 넘길수도 있어요!
> react-router에 있는 useHistory와 useLocation에 대한 정보를 찾아보면 좋을거같아요
> 
> ```
> // 예제
> history.push({
>             pathname: '/cardComplete',  // 이동할 페이지 라우터
>             search: '?query=userName',  // 에 넣어줄 쿼리
>             state: { cardFormInfo: cardFormInfo },  // 넘겨줄 state
>           });
> ...
> ```

질문있습니다! 
router로 넘길때 history의 state로 넘겨준다면 보안 이슈는 없을까요?
카드번호, 비밀번호 등은 민감한 정보인데, 브라우져의 history API는 악성 유저로부터 탈취당할 가능성은 없을까요..?
